senders/mattermost/sender.go

```diff
@@ -0,0 +1,187 @@
+package mattermost
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/moira-alert/moira"
+	"github.com/moira-alert/moira/senders"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+)
+
+// Sender posts messages to Mattermost chat.
+// It implements moira.Sender.
+// You must call Init method before SendEvents method.
+type Sender struct {
+	frontURI string
+	location *time.Location
+	client   Client
+}
+
+// Init configures Sender.
+func (sender *Sender) Init(senderSettings map[string]string, _ moira.Logger, location *time.Location, _ string) error {
+	url := senderSettings["url"]
+	if url == "" {
+		return fmt.Errorf("can not read Mattermost url from config")
+	}
```

Comment:
```suggestion
	url, ok := senderSettings["url"]
	if !ok {
		return fmt.Errorf("can not read Mattermost url from config")
	}
```

Reply:
Current code covers both cases: key doesn't exists (Go map returns zero value for value if key doesn't exists, so it is empty string) or value is empty string. I think we should return error in both cases. Thank you, I added some tests to this case https://github.com/moira-alert/moira/pull/791/commits/4634bd6d445f563064fc5ed973578b1a8deb3fbd